src/tools/perf/libperf.c

```diff
@@ -1012,3 +1027,106 @@ ucs_status_t ucx_perf_run(ucx_perf_params_t *params, ucx_perf_result_t *result)
 out:
     return status;
 }
+
+
+/* multiple threads sharing the same worker/iface */
+
+typedef struct {
+    pthread_t pt;
+    int tid;
+    int ntid;
+    ucx_perf_context_t perf;
+    ucx_perf_params_t params;
+    ucx_perf_result_t result;
+} ucx_perf_thread_context_t;
+
+static ucs_status_t* ucx_perf_thread_statuses;
+static pthread_barrier_t ucx_perf_thread_barrier;
+
+static void* ucx_perf_thread_run_test(void* tctx_) {
```

Comment:
please avoid using trailing _ in the name, can use `void *arg` as the param name